test/puppeteer/docs.test.ts

```diff
@@ -1,11 +1,63 @@
 // Imports
 import path from 'path';
 
+// Destructured constant for readability
+const { keyboard } = page;
+
+// Helper functions used in multiple tests
+const menuDisplayState = () => page.evaluate(() => document.querySelector<HTMLDivElement>('#menu').style.display);
+const currentFocusId = () => page.evaluate(() => document.activeElement.id);
+
 // Tests
-beforeAll(async () => {
+beforeEach(async () => {
 	await page.goto(`file://${path.join(__dirname, '..', '..', 'docs', 'build', 'index.html')}`);
+	await page.waitFor('#menubutton');
 });
 
-it('should have correct page title', async () => {
+it('has the correct page title', async () => {
 	await expect(page.title()).resolves.toMatch('React Accessible Dropdown Menu Hook');
 });
+
+it('focuses on the first menu item when the enter key is pressed', async () => {
+	await page.focus('#menubutton');
+	await keyboard.down('Enter');
+
+	expect(await currentFocusId()).toBe('menuitem1');
+});
+
+it('hides the menu by default and make it visible if it is open', async () => {
```

Comment:
Isn't it up to the component that uses this Hook to actually show/hide the menu? These tests should focus on exercising the responsibility of our Hook rather than the responsibility of the client component.

Reply:
It's up to the hook to return the state of the menu being opened or closed, which enzyme could not accurately test. This was the best indirect way I could come up with to make sure the value returned by the hook was correct, since the styling of the menu is based on it. Maybe a rewording could be used? I do think the "isOpen" return value of the hook needs to be tested in some way.

Reply:
I think we should do this type of test in Enzyme, albeit indirectly. What if the test component we used during the Enzyme tests contained a `<span />` tag like:

```tsx
<span id='is-open-indicator'>{isOpen ? 'true' : 'false'}</span>
```

Could we use Enzyme to get the content of `span#is-open-indicator`?

Reply:
That's a workable solution, I'll remove the tests since we plan to do that.